NDTensors/ext/NDTensorsMetalExt/permutedims.jl

```diff
@@ -1,3 +1,10 @@
+function Base.permutedims!(
+  Edest::Unwrap.Exposed{<:MtlArray}, Esrc::Unwraped.Exposed{<:MtlArray}, perm
```

Comment:
Probably needs a more specific signature here since this shouldn't be used for all `MtlArray` operations, i.e. specialize on what wrapper type was giving trouble (I think `ReshapedArray`)?

Reply:
Makes sense! I knew there would be some issues with the metal implementation I just wasn't able to check on my workstation. I'm on my mac today and will verify this code. Thanks!

Reply:
Ahh yes I see in the next line down you have `Adest::Base.ReshapedArray{<:Any,<:Any,<:SubArray}`  so I'll make `Edest::Exposed{<:MtlArray, <:ReshapedArray}` and will run tests to see if that works

Reply:
So probably it should be `Edest::Exposed{<:MtlArray,ReshapedArray{<:Any,<:Any,<:SubArray}}`, I don't think there was an issue with all `ReshapedArray`, just ones wrapping a `SubArray`.